### PR TITLE
use lapack and numpy==1.21.2 on ARM

### PR DESCRIPTION
**Pull request by heidongxianhua**
as title, fix the pytorch nightly build, https://github.com/pytorch/pytorch/issues/104950

### CODE REVIEW

File: aarch64_linux/aarch64_ci_build.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ PATH=/opt/conda/bin:$PATH
 # Install OS dependent packages
 ###############################################################################
 yum -y install epel-release
-yum -y install less zstd libgomp
+yum -y install less zstd libgomp liblapack-dev
```
Comment (xncqr) [inline aarch64_linux/aarch64_ci_build.sh:15]: Believe  liblapack-dev is for "Debian" based distros. Think you need lapack-devel from "RHEL" based distros as this is using pypi manylinux docker image to build.
ref: https://centos.pkgs.org/7/centos-x86_64/lapack-devel-3.4.2-8.el7.x86_64.rpm.html

**Comment by xncqr** [comment]
@heidongxianhua,

Do not know whether you tested this with the pypi manylinux build. I have test ENV for the work I have done on aarch64 and am testing the change on that ENV.

Note: I am not one of the maintainers of PyTorch and would not be able to approve this PR. Will see if I can find someone after I can verify this is good for manylinux.

**Comment by xncqr** [comment]
@heidongxianhua,

don't know if you tested this but LAPACK did not get built into the wheel when "only" installing the yum package. Looks to be only Ubuntu where that operation worked.

**Comment by heidongxianhua** [comment]
> nstallin

yeah, I do not have the right machine to test it now. But I check the log in nightly build CI work, https://github.com/pytorch/pytorch/actions/runs/5516843118/jobs/10058689778, in the docker there has `liblapack-dev` package 
```
The following NEW packages will be INSTALLED:

  boltons            conda-forge/noarch::boltons-23.0.0-pyhd8ed1ab_0 
  jsonpatch          conda-forge/noarch::jsonpatch-1.32-pyhd8ed1ab_0 
  jsonpointer        conda-forge/noarch::jsonpointer-2.0-py_0 
  libblas            conda-forge/linux-aarch64::libblas-3.9.0-17_linuxaarch64_openblas 
  libcblas           conda-forge/linux-aarch64::libcblas-3.9.0-17_linuxaarch64_openblas 
  libgfortran-ng     conda-forge/linux-aarch64::libgfortran-ng-13.1.0-he9431aa_0 
  libgfortran5       conda-forge/linux-aarch64::libgfortran5-13.1.0-h24e4805_0 
  libgit2            conda-forge/linux-aarch64::libgit2-1.6.4-h2bc94ca_0 
  liblapack          conda-forge/linux-aarch64::liblapack-3.9.0-17_linuxaarch64_openblas 
```
But when build the torch wheel , the cmake summary info is:
```
2023-07-11T07:26:01.2040848Z --   USE_BLAS              : 1
2023-07-11T07:26:01.2041220Z --     BLAS                : open
2023-07-11T07:26:01.2041607Z --     BLAS_HAS_SBGEMM     : 
2023-07-11T07:26:01.2041995Z --   USE_LAPACK            : 0
```
So I think we also have to set the ENV `USE_LAPACK=1`

**Comment by heidongxianhua** [comment]
> even setting USE_LAPACK=1 does not have it build with LAPACK. need to solve the incompatibility with the manylinux2014 image (worked building on Ubuntu). This will not work at the moment. Am looking into it. Will take a bit.

yes, thanks for your tests. And another question, does it will use the same docker and setting to build the torch release version of 2.1.0？

**Comment by xncqr** [comment]
@heidongxianhua,

Not sure that you are asking. if you are asking if the same docker image is used for the builds, yes. aarch64's build base image is aarch64 manylinux_2014. I'm working on a pytorch builder image, in line with x86 build process and have that under test. I updated your issue on why manylinux is used.

Also, you can probably close this PR. Once I have a process that is adding LAPACK using manylinux image, I will be submitting a PR to resolve, along with some other updates for the build process (likely include the aarch64 pytorch builder docker image process so conda and sub packages can be built)

**Pull request closed without merging**